### ISSUE TITLE
Unexpected error with opa eval --target=wasm

### ISSUE THREAD

**Issue by tsandall** [open-policy-agent/opa#3263]
There's something wrong in `opa eval` that's preventing it from handling certain types of queries. Here's a simple example:

```
$ cat ~/x.json
[1]
```

```
torin:~$ opa eval -i ~/x.json -t wasm 'input[x]==1'
{
  "errors": [
    {
      "message": "expected exactly one term but got: "
    }
  ]
}
torin:~$ opa eval -i ~/x.json -t rego 'input[x]==1'
{
  "result": [
    {
      "expressions": [
        {
          "value": true,
          "text": "input[x]==1",
          "location": {
            "row": 1,
            "col": 1
          }
        }
      ],
      "bindings": {
        "x": 0
      }
    }
  ]
}
```

**Comment by srenatus** [comment]
👉 for reference, this had come up here, too: https://github.com/open-policy-agent/opa/issues/3128#issuecomment-774984367

**Comment by srenatus** [comment]
```
$ ./opa_linux_amd64 eval -t wasm 'input[x] == 1' --input input.json
{
  "result": [
    {
      "expressions": [
        {
          "value": true,
          "text": "input[x] == 1",
          "location": {
            "row": 1,
            "col": 1
          }
        }
      ],
      "bindings": {
        "x": 0
      }
    }
  ]
}

```

😄 Fixed as another side-effect of #3271.
